addons/web/static/src/js/fields/field_utils.js

```diff
@@ -203,24 +203,69 @@ function formatFloatFactor(value, field, options) {
  * @param {Object} [field]
  *        a description of the field (note: this parameter is ignored)
  * @param {Object} [options]
- * @param {boolean} [options.noLeadingZeroHour] if true, format like 1:30
- *        otherwise, format like 01:30
+ * @param {string} [unit] gives the unit of the original float in hours, minutes or seconds.
+ *        The default is 'hours'. (/!\ Do not precise the units as minutes if you don't want to change the default pattern)
+ *        Note: If the float must convert to the default xx:yy, no need to specify if those are minutes or hours.
+ * @param {string} [pattern] Allow any custom pattern to be used.
+ *         - Put the asked datas into brackets (e.i. '[D]')
+ *         - One character will have no leading zero ('[H]h' => 1h)
+ *         - Two characters will have a leading zero ('[HH]'h => 01h)
+ *         - Write any sentence to format ('Wow this took [D] days!' => Wow this took 2 days!)
+ *         - D = days
+ *           H = hours
+ *           M = minutes
+ *           S = seconds
+ *         - The DEFAULT patterns will be (depending on the relevant infos):
+ *           mm:ss || hh:mm (/!\ Do not precise the units as minutes if you don't want to change this pattern)
```

Comment:
I would rename it 'format' to align with every other formatting brols.
```suggestion
 * @param {string} [pattern] Allow any custom formatting to be used.
 *         - Desired time unit are put into bracket, separated by time separator (':', '-', ...). E.g. : "[D]:[HH]:[MM]:[SS]"
 *         - One character will have no leading zero ('[H]' => 1h)
 *         - Two characters will have a leading zero ('[HH]' => 01h)
 *         - time units : 
 *           D = days
 *           H = hours
 *           M = minutes
 *           S = seconds
 *         - The DEFAULT patterns will be [HH:MM]
```
for the default, I don't understand. It's depending on what exactly ? but anyway, you cannot have two default 'patterns'. I vote for [HH]:[MM] as it seems that it was the original 'pattern', and because it's what the code actually does.

Regarding allowing sentence to come with the arguments: not a good idea -> This would need to handle translation.
If someone wants to make a sentence, them will do it manually in the template and just insert the float_time in the right place. like :
```
<div>
    <span>Wow, this took
    <span class="font-weight-bold" field name="average_duration" widget="float_time" options="{'unit':'seconds', 'pattern':'[D]'}"</span>
    day(s)</span>
</div>
```

Reply:
@dbeguin actually, currently, the default behavior is already hh:mm or mm:ss. This didn't change and should stay that way to ensure compatibility with what already exists.

Ok I'll change the format to better match other formatting. I could still keep that custom behavior to allow things like 
01'30 || 01:30 || 5-01:30:45 if we want

Reply:
Regarding 'format' , this seems to be a reserved naming in basic_field to format numeric fields. 
So keep pattern.